### DISCUSSION TITLE
Staring the service with different initial state

### DISCUSSION THREAD

**Discussion by pernilsalat** [statelyai/xstate#2151] [Q&A] [answered]
Let's say i have the following machine: 
```            
           {
              id: 'toggle',
              initial: 'active',
              states: {
                active: {
                  on: {
                    TOGGLE: { target: 'inactive' },
                  },
                },
                inactive: {
                  on: {
                    TOGGLE: { target: 'active' },
                  },
                },
              },
            }
```
Is there a way to start the service in the `inactive` state?

As far as I have read in the docs, you can do something like that:
```
service.start(State.create(JSON.parse(persistedStringifyState)))
```

The thing is that i would like to persist only the state name (ex. `active`/`inactive`)

So i have tried something like the following code 
```
service.start(State.create(service.machine.states.inactive))
```
or
```
service.start(service.machine.states.inactive)
```
without much success  (`cannot read data of undefined`)

So is there a way to achieve what i want? Sorry if this have been asked, i could not find that discussion so i made this one
Thanks in advance!

**Comment by RareSecond** [comment]
Hi @pernilsalat, I think that a [transient state node](https://xstate.js.org/docs/guides/statenodes.html#transient-state-nodes) might be exactly what you're looking for!

**Reply by pernilsalat** [reply]
But this would make me change the configuration (and adding a lot of boilerplate) of the machine just to compensate this setup problem that I am having (if i understand it correctly), thing that i don't think it is a good practice.

Am i understanding it correctly?

**Comment by davidkpiano** [comment, marked as answer]
Try `service.start('inactive')`

**Reply by pernilsalat** [reply]
Of course it works... Shame on me for trying everything except the most simple thing.

Thank you very much!!
